### PR TITLE
[ROCm 6.0.1] Adaptation for HIPRTC changes in 6.0.1

### PR DESCRIPTION
**Pull request by atamazov**
- 7e8844940 regression: do not use file system symbolic/hard links (cherry-picked from [#2425](https://github.com/ROCm/MIOpen/pull/2425))
- c4a091f71 Patch necessary to make FP8 convolution compile with hiprtc (cherry-picked from [#2584](https://github.com/ROCm/MIOpen/pull/2584))
- 2c9fe9689 Bump MIOpen version to 3.0.1 and update CI docker (cherry-picked from [#2519](https://github.com/ROCm/MIOpen/pull/2519) and EDITED)
  - Differences from [#2519](https://github.com/ROCm/MIOpen/pull/):
    - rocm_setup_version(VERSION 3.1.0) -> rocm_setup_version(VERSION 3.0.1)
    - Changes of requirements.txt REVERTED
- c3de5e457 [ROCm 6.0.1][hipRTC] Fix build failures. [quality] Reorg standard includes in HIP sources. (cherry-picked from [#2637](https://github.com/ROCm/MIOpen/pull/2637) and EDITED)
  - Differences from [#2637](https://github.com/ROCm/MIOpen/pull/2637):
    - Almost all changes of test/gtest/CMakeLists.txt reverted, except:
      - extending timeout for gtest discovery
      - logging more testing parameters onto console
- 4f695d975 Automatically activate the new HIPRTC PCH adaptations starting from the 6.0.24000. Fix some build errors.
  - ℹ️ You can use `-DMIOPEN_OVERRIDE_HIP_VERSION_PATCH=XXXX` to control this behaviour.
    - For example, if some future ROCm RC, say, 2.0.241111, contains old HIPRTC, then you may need to disable the new HIPRTC adaptations. This can be done by `-DMIOPEN_OVERRIDE_HIP_VERSION_PATCH=23999` when building MIOpen.
    - Similarly, if you want to test MIOpen with some, for example. ROCm RC 6.0.23525, and that ROCm needs the new HIPRTC adaptations, then you can use `-DMIOPEN_OVERRIDE_HIP_VERSION_PATCH=24000`.

### ⚠️ If this PR works fine, then we'll have to cherry-pick commit 4f695d975 into `develop`.

- That would maintain compatibility of future MIOpen releases with ROCm 6.0.1.
- Please expect merge conflicts (the trivial ones, though) during cherry-picking because this commit is partially based on [#2465](https://github.com/ROCm/MIOpen/pull/2465) which has been already merged into `develop`.

## :cyclone: Testing

- Navi21 (gfx1030/36), Ubuntu 20.04.5 LTS
- Docker image: `rocm/miopen:hiprtc` with dependencies installed from `requirements.txt` from this PR.
- `-DMIOPEN_TEST_ALL=Off -DBUILD_DEV=On` (Smoke testing)
- `-DMIOPEN_OVERRIDE_HIP_VERSION_PATCH=2400`
- Tested builds, compilation paths & datatypes:
  -  FLOAT, offline compiler, release build
  -  FLOAT, online compiler, DEBUG build
  -  HALF, online compiler, release build
  -  BFLOAT16, online compiler, release build
  -  INT8, online compiler, release build

🟢 ALL PASSED

----
[Attribution] @junliume @JehandadKhan
- https://github.com/ROCm/MIOpen/labels/enhancement
- https://github.com/ROCm/MIOpen/labels/urgency_unknown
- https://github.com/ROCm/MIOpen/labels/value_high
- Proposed assignees:
- Proposed reviewers:
  - @JehandadKhan
  - @junliume
  - @CAHEK7

### CODE REVIEW

**Comment by atamazov** [comment]
@junliume I see issues with some smoke_solver_ConvHipImplicitGemm* tests and with LayerNormTestSet. Investigating...

**Comment by atamazov** [comment]
@junliume This is ready to go.

**Review by CAHEK7 — APPROVED** [review]

**Review by junliume — APPROVED** [review]

**Comment by atamazov** [comment]
@junliume Excuse me, but I do not understand why this branch has been squashed on merge.

**Comment by junliume** [comment]
> @junliume Excuse me, but I do not understand why this branch has been squashed on merge.

Ops, used default setting. The Cherry Pick history is still visible from the squashed commit message though. The staging has already started, I will see if there is a good way to "unsquash" the commits.

**Comment by atamazov** [comment]
@junliume No problem for me. I cherry-picked #2644 from my development branch. Let's hope the maintainers of the release branch won't have any problems either.

File: src/kernels/miopen_limits.hpp [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+#pragma once
+
+#ifndef WORKAROUND_DO_NOT_USE_CUSTOM_LIMITS
+#define WORKAROUND_DO_NOT_USE_CUSTOM_LIMITS 0
+#endif
```
Comment (atamazov) [inline src/kernels/miopen_limits.hpp:30]: @junliume Another instance of the quality issue. Better to fix, but not mandatory.

**Comment by atamazov** [comment]
@junliume Does this work correctly with ROCm 6.0.1 RC? If everything is fine in the release staging branch, then let's merge #2644.

**Comment by junliume** [comment]
> @junliume Does this work correctly with ROCm 6.0.1 RC? If everything is fine in the release staging branch, then let's merge #2644.

Still waiting for the staging results as of 01/03

**Pull request merged**